### PR TITLE
Update samples to 0.74

### PR DESCRIPTION
**Pull request by Yajur-Grover**
## Description
Upgrade Calculator, CalculatorNuget and NativeModule samples to 0.74 as part of 0.74 release.

### Why

Part of https://github.com/microsoft/react-native-windows/issues/12874

## Screenshots
N/A
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/react-native-windows-samples/pull/937)

### CODE REVIEW

File: samples/Calculator/csharp/package.json [hunk 1]
```diff
@@ -12,17 +12,17 @@
   },
   "dependencies": {
     "react": "18.2.0",
-    "react-native": "0.73.0",
-    "react-native-windows": "0.73.2"
+    "react-native": "0.74.0-rc.9",
```
Comment (chrisglein) [inline samples/Calculator/csharp/package.json:15]:
```suggestion
    "react-native": "0.74.0",
```
Assuming @TatianaKapos validated that this works fine.

File: samples/Calculator/csharp/windows/calculator/Package.appxmanifest [hunk 2]
```diff
@@ -7,15 +7,15 @@
...
     Publisher="CN=React Native Windows Sample"
     Version="1.0.0.0" />
 
-  <mp:PhoneIdentity PhoneProductId="b3e4af42-aec3-46e8-a138-cb2e481514b2" PhonePublisherId="00000000-0000-0000-0000-000000000000"/>
+  <mp:PhoneIdentity PhoneProductId="7657a35a-6073-4280-b14b-409220cbd347" PhonePublisherId="00000000-0000-0000-0000-000000000000"/>
 
   <Properties>
     <DisplayName>Calculator</DisplayName>
-    <PublisherDisplayName>CN=React Native Windows Sample</PublisherDisplayName>
+    <PublisherDisplayName>yajurgrover</PublisherDisplayName>
```
Comment (chrisglein) [inline samples/Calculator/csharp/windows/calculator/Package.appxmanifest:18]: Revert
```suggestion
    <PublisherDisplayName>CN=React Native Windows Sample</PublisherDisplayName>
```

File: samples/Calculator/csharp/windows/calculator/Package.appxmanifest [hunk 3]
```diff
@@ -7,15 +7,15 @@
   IgnorableNamespaces="uap mp">
 
   <Identity
-    Name="b3e4af42-aec3-46e8-a138-cb2e481514b2"
+    Name="7657a35a-6073-4280-b14b-409220cbd347"
     Publisher="CN=React Native Windows Sample"
     Version="1.0.0.0" />
 
-  <mp:PhoneIdentity PhoneProductId="b3e4af42-aec3-46e8-a138-cb2e481514b2" PhonePublisherId="00000000-0000-0000-0000-000000000000"/>
+  <mp:PhoneIdentity PhoneProductId="7657a35a-6073-4280-b14b-409220cbd347" PhonePublisherId="00000000-0000-0000-0000-000000000000"/>
```
Comment (chrisglein) [inline samples/Calculator/csharp/windows/calculator/Package.appxmanifest:14]: Ideally the package identity is stable across updates. If you create from a fresh app template you'll need to copy the project and package ids back over.

File: samples/Calculator/csharp/windows/calculator/calculator.csproj [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
-<!-- This project was created with react-native-windows 0.73.2 -->
+<!-- This project was created with react-native-windows 0.74.0 -->
 <Project ToolsVersion="Current" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
   <Import Project="$(SolutionDir)\ExperimentalFeatures.props" Condition="Exists('$(SolutionDir)\ExperimentalFeatures.props')" />
   <PropertyGroup>
     <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
     <Platform Condition=" '$(Platform)' == '' ">x86</Platform>
-    <ProjectGuid>{7410B9FC-308E-4496-B556-FC56F1910AA7}</ProjectGuid>
+    <ProjectGuid>{B2ABB10D-1A83-4DF1-8D0C-7167F3F88A5F}</ProjectGuid>
```
Comment (chrisglein) [inline samples/Calculator/csharp/windows/calculator/calculator.csproj:9]: You should be able to do a find replace to put back the old project guide.

File: samples/CalculatorNuGet/windows/calculatornuget/Package.appxmanifest [hunk 5]
```diff
@@ -7,15 +7,15 @@
...
     Publisher="CN=React Native Windows Sample"
     Version="1.0.0.0" />
 
-  <mp:PhoneIdentity PhoneProductId="b36753c3-2e37-4dd8-8d3d-66953d899343" PhonePublisherId="00000000-0000-0000-0000-000000000000"/>
+  <mp:PhoneIdentity PhoneProductId="bad696f0-699e-47a2-87df-2877b64a81a6" PhonePublisherId="00000000-0000-0000-0000-000000000000"/>
 
   <Properties>
     <DisplayName>CalculatorNuGet</DisplayName>
-    <PublisherDisplayName>CN=React Native Windows Sample</PublisherDisplayName>
+    <PublisherDisplayName>yajurgrover</PublisherDisplayName>
```
Comment (chrisglein) [inline samples/CalculatorNuGet/windows/calculatornuget/Package.appxmanifest:18]: As with other project, leave the package and project information intact.

**Review by chrisglein — CHANGES_REQUESTED** [review]

File: samples/Calculator/csharp/package.json (hunk 1 above)
Comment (Yajur-Grover) [reply samples/Calculator/csharp/package.json:15]: Thought I reverted them all - will fix!

File: samples/Calculator/csharp/windows/calculator/Package.appxmanifest (hunk 3 above)
Comment (Yajur-Grover) [reply samples/Calculator/csharp/windows/calculator/Package.appxmanifest:14]: Is this for both `Name` and `PhoneIdentity`? Or just `PhoneIdentity`?

File: samples/Calculator/csharp/windows/calculator/calculator.csproj (hunk 4 above)
Comment (chrisglein) [reply samples/Calculator/csharp/windows/calculator/calculator.csproj]: FYI the project guid is just a Visual Studio thing. But it should be able to be left unchanged.
The one in the package manifest is what actually determines the store listing. So changing that is more disruptive.

File: samples/Calculator/csharp/windows/calculator/Package.appxmanifest [hunk 6]
```diff
@@ -15,7 +15,7 @@
 
   <Properties>
     <DisplayName>Calculator</DisplayName>
-    <PublisherDisplayName>CN=React Native Windows Sample</PublisherDisplayName>
+    <PublisherDisplayName>"CN=React Native Windows Sample"</PublisherDisplayName>
```
Comment (chrisglein) [inline samples/Calculator/csharp/windows/calculator/Package.appxmanifest:18]:
```suggestion
    <PublisherDisplayName>CN=React Native Windows Sample</PublisherDisplayName>
```

**Review by chrisglein — APPROVED** [review]

File: samples/NativeModuleSample/csharp/windows/NativeModuleSample/packages.lock.json [hunk 7]
```diff
@@ -139,7 +139,10 @@
         "contentHash": "qF6RRZKaflI+LR1YODNyWYjq5YoX8IJ2wx5y8O+AW2xO+1t/Q6Mm+jQ38zJbWnmXbrcOqUYofn7Y3/KC6lTLBQ=="
       },
       "common": {
-        "type": "Project"
+        "type": "Project",
+        "dependencies": {
+          "boost": "[1.76.0, )"
+        }
```
Comment (Yajur-Grover) [inline samples/NativeModuleSample/csharp/windows/NativeModuleSample/packages.lock.json:145]: Not sure about this change - didn't create any issues when building the sample but wanted to confirm if this should be kept.

File: samples/Calculator/cppwinrt/README.md [hunk 8]
```diff
@@ -4,7 +4,7 @@ See [../README.md](../README.md) for details of this sample.
 
 See [../csharp/](../csharp/) for a C# version of this sample.
 
-It currently targets React Native Windows 0.73.
+It currently targets React Native Windows 0.74.
```
Comment (TatianaKapos) [inline samples/Calculator/cppwinrt/README.md:7]: Can you also update the ReadMe in the parent folder? (also maybe add that to the ReadMe upgrade instructions, I also always forgot to update that one!)

File: samples/NativeModuleSample/csharp/windows/NativeModuleSample/packages.lock.json (hunk 7 above)
Comment (TatianaKapos) [reply samples/NativeModuleSample/csharp/windows/NativeModuleSample/packages.lock.json:145]: May just be from a package dependency, if it didn't break the project it's fine to keep :)

**Review by TatianaKapos — APPROVED** [review]

**Review by chiaramooney — APPROVED** [review]

**Pull request merged**